nautobot/dcim/forms.py

```diff
@@ -3235,8 +3232,15 @@ def __init__(self, *args, **kwargs):
             # Limit VLAN choices by Location
             if locations.count() == 1:
                 location = locations.first()
-                self.fields["untagged_vlan"].widget.add_query_param("locations", location.pk)
+                # In the case of a single location, use the available_on_device query param to limit untagged VLAN choices
+                # to those available on the devices in that location and in the ancestors of the location.
+                device_pks = [device.pk for device in devices]
+                self.fields["untagged_vlan"].widget.add_query_param("available_on_device", device_pks)
                 self.fields["tagged_vlans"].widget.add_query_param("locations", location.pk)
+            else:
+                self.fields["untagged_vlan"].widget.add_query_param("locations", "null")
+                if locations.count() > 1:
+                    self.fields["untagged_vlans"].widget.add_query_param("locations", list(locations.values_list("pk", flat=True)))
```

Comment:
Should I account for the ancestors of these locations as well?

Reply:
Probably should... does InterfaceEditForm do so now though?

Reply:
`available_on_device` is the filter that we use on InterfaceEditForm and that includes the ancestors. But I do not think this code block matters because I think the `locations.count()` can only be 0 or 1

Reply:
I'm confused then - if you're editing interfaces from N different devices, can't there be N locations?